tests/test_setup_kubernetes_job.py

```diff
@@ -51,78 +45,45 @@ def test_main():
             soa_dir=mock_parse_args.return_value.soa_dir,
         )
-
         mock_setup_kube_deployments.return_value = False
         with raises(SystemExit) as e:
             main()
         assert e.value.code == 1
 
 
-def test_setup_kube_deployment():
+def test_setup_kube_deployment_invalid_job_name():
     with mock.patch(
-        "paasta_tools.setup_kubernetes_job.reconcile_kubernetes_deployment",
-        autospec=True,
-    ) as mock_reconcile_kubernetes_deployment, mock.patch(
+        "paasta_tools.setup_kubernetes_job.create_application_object", autospec=True
+    ) as mock_create_application_object, mock.patch(
         "paasta_tools.setup_kubernetes_job.list_all_deployments", autospec=True
     ) as mock_list_all_deployments:
         mock_client = mock.Mock()
-        mock_service_instances: Sequence[str] = []
-        assert (
-            setup_kube_deployments(
-                kube_client=mock_client,
-                service_instances=mock_service_instances,
-                soa_dir="/nail/blah",
-            )
-            is True
-        )
-
-        mock_reconcile_kubernetes_deployment.return_value = (0, 0)
-        mock_service_instances = ["kurupt.fm", "kurupt.garage"]
-        assert (
-            setup_kube_deployments(
-                kube_client=mock_client,
-                service_instances=mock_service_instances,
-                soa_dir="/nail/blah",
-            )
-            is True
-        )
-        mock_reconcile_kubernetes_deployment.assert_has_calls(
-            [
-                mock.call(
-                    kube_client=mock_client,
-                    service="kurupt",
-                    instance="fm",
-                    kube_deployments=mock_list_all_deployments.return_value,
-                    soa_dir="/nail/blah",
-                ),
-                mock.call(
-                    kube_client=mock_client,
-                    service="kurupt",
-                    instance="garage",
-                    kube_deployments=mock_list_all_deployments.return_value,
-                    soa_dir="/nail/blah",
-                ),
-            ]
-        )
-
-        mock_reconcile_kubernetes_deployment.return_value = (1, 0)
-        assert (
-            setup_kube_deployments(
-                kube_client=mock_client,
-                service_instances=mock_service_instances,
-                soa_dir="/nail/blah",
+        mock_list_all_deployments.return_value = [
+            KubeDeployment(
+                service="kurupt", instance="f_m", git_sha="", config_sha="", replicas=0
             )
-            is False
+        ]
+        mock_service_instances = ["kuruptf_m"]
+        setup_kube_deployments(
+            kube_client=mock_client,
+            service_instances=mock_service_instances,
+            soa_dir="/nail/blah",
         )
+        assert mock_create_application_object.call_count == 0
 
 
 def test_ensure_pod_disruption_budget_create():
```

Comment:
Since the code has moved to controller_wrappers, the tests for pod disruption budgets should go there too.